### PR TITLE
Small macro cleanups

### PR DESCRIPTION
**Pull request by kmcallister**
r? @alexcrichton. This passes tests for me.

### CODE REVIEW

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -60,184 +60,6 @@ macro_rules! panic {
...
-/// ```
-#[macro_export]
-#[stable]
-macro_rules! assert {
-    ($cond:expr) => (
-        if !$cond {
-            panic!(concat!("assertion failed: ", stringify!($cond)))
-        }
-    );
-    ($cond:expr, $($arg:tt)+) => (
```
Comment (alexcrichton) [inline src/libstd/macros.rs]: this use of `:tt` wasn't actually propagated down to libcore, could you do that as well?

File: src/libstd/macros.rs [hunk 2]
```diff
@@ -60,184 +60,6 @@ macro_rules! panic {
...
-/// assert!(some_computation());
-///
-/// // assert with a custom message
-/// # let x = true;
-/// assert!(x, "x wasn't true!");
-/// # let a = 3i; let b = 27i;
-/// assert!(a + b == 30, "a = {}, b = {}", a, b);
-/// ```
-#[macro_export]
-#[stable]
```
Comment (alexcrichton) [inline src/libstd/macros.rs]: Can you also propagate these `stable` attributes downwards as well?

**Comment by alexcrichton** [comment]
Nice thinking! I forgot we could rely on `panic!` being redefine in libstd.

Could you basically do a pass where you copy-paste the current definitions into libcore? They've been tweaked slightly and just want to make sure all the changes stay in place.

Thanks!

**Comment by kmcallister** [comment]
Pushed the libcore updates, along with a rebase.

> Nice thinking! I forgot we could rely on `panic!` being redefine in libstd.

Yeah, this basically exploits a lack of hygiene on macros referring to other macros. Which I don't think we'll fix any time soon...

**Pull request merged**